src/App.vue

```diff
@@ -249,12 +249,6 @@ export default {
     },
 
     zazzleMugPrint() {
-      if (this.zazzleLink) {
-        window.open(this.zazzleLink, '_blank');
-        recordOpenClick(this.zazzleLink);
-        return;
```

Comment:
Unfortunately this code is necessary due to security restrictions in some browsers. 

During handling of the print request, we upload image to an s3 bucket, so that zazzle can fetch it for rendering. Only after upload is done can we open a zazzle window with correct link to the image.

Most browsers do not allow to open new windows unless action was explicitly initiated by a user (e.g. click) on the same event loop cycle. So, if browser has blocked the window, `this.zazzleLink` will point to the right location, and subsequent click will be synchronous, allowing us to open the window.

That was the reason why this architecture is in place. However you are right - it doesn't make sense to keep the link stale. I've pushed/deployed [code to properly reset](https://github.com/anvaka/city-roads/commit/afc14d41b2c6143dd9654627bc299ebe8e85eb3f) `this.zazzleLink` in response to changes of the canvas.

Reply:
Right, this explains it, I did overlook the pop up restriction. 
Opening a window right away and then changing it's location works as a workaround, but doesn't feel right, especially if s3 request fails and one has to close the newly opened window.

I appreciate you going an extra mile doing it the right way which I tried to avoid for keeping it simple.